web/src/search/results/SearchResultTab.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react'
+import * as H from 'history'
+import { SEARCH_TYPES } from './SearchResults'
+import { NavLink } from 'react-router-dom'
+import { toggleSearchType } from '../helpers'
+import { buildSearchURLQuery } from '../../../../shared/src/util/url'
+
+interface Props {
+    location: H.Location
+    history: H.History
+    type: SEARCH_TYPES
+    query: string
+}
+
+const typeToProse: Record<SEARCH_TYPES, string> = {
+    '': 'Code',
+    diff: 'Diffs',
+    commit: 'Commits',
+    symbol: 'Symbols',
+    repo: 'Repos',
+}
+
+const tabIsActive = (builtQuery: string, location: H.Location): boolean => location.search === '?' + builtQuery
+
+const tabIsActiveTrue = (): boolean => true
+const tabIsActiveFalse = (): boolean => false
+
+export const SearchResultTab: React.FunctionComponent<Props> = props => {
+    const q = toggleSearchType(props.query, props.type)
+    const newURLSearchParam = buildSearchURLQuery(q)
+
+    const isActiveFunc = tabIsActive(newURLSearchParam, props.location) ? tabIsActiveTrue : tabIsActiveFalse
```

Comment:
I suggest using lodash's `constant` for this:

```suggestion
    const isActiveFunc = constant(location.search === `?${builtQuery}`)
```